sbt/Dockerfile

```diff
@@ -20,7 +21,16 @@ ARG BASE_URL=https://sbt.bintray.com/rpm/
 RUN curl -fsSL -o /tmp/sbt-${TOOL_VERSION}.rpm ${BASE_URL}/sbt-${TOOL_VERSION}.rpm && \
   yum install -y /tmp/sbt-${TOOL_VERSION}.rpm && \
   yum clean all
-RUN echo "-Dsbt.override.build.repos=true" >> /etc/sbt-launcher-packaging/sbtopts
+
+# The location of global sbtopts is dependent on the version of SBT (rpm) being installed
+# As of 2018/07/24, there are 29 versions of the SBT rpm available on bintray
+# 10 versions use /usr/share/sbt-launcher-packaging/conf    (which is symlinked to /etc/sbt-launcher-packaging)
+# 19 versions use /usr/share/sbt/conf                       (which is symlinked to /etc/sbt)
+# So check for the existence of sbtopts in two locations before appending to it.
```

Comment:
See https://gist.github.com/ajorpheus/9ce9ed0a2639d1494d1649b4624d9bcd for details of the versions and the location of sbtopts

Reply:
Nice fix :+1:

Reply:
I tried a bunch of things to make the fix more scalable (for instance, what happens if we have to put in more global defaults in sbtopts. I wanted to export the path of the sbtopts in an ENV var and then append to that, but docker was not having any of it.

```
ENV SBTOPTS_PATH="/dev/null"
RUN if [ -f /etc/sbt-launcher-packaging/sbtopts ]; then export $SBTOPTS_PATH="/etc/sbt-launcher-packaging/sbtopts" ; \
  elif [ -f /etc/sbt/sbtopts ]; then export $SBTOPTS_PATH="/etc/sbt/sbtopts" ; \
  else exit 1; fi

RUN echo "-Dsbt.override.build.repos=true" >> $SBTOPTS_PATH
```

Also tried a different approach creating an sbt symlink which will always be present. But that did not work either.

)

Reply:
I think ideally we would want to put the whole file into the container at build time with our config - but this will most likely result in maintaining multiple versions. I guess thats a complexity we adopt with SBT.